src/datasources/balances-api/coingecko-api.service.ts

```diff
@@ -92,12 +100,12 @@ export class CoingeckoApi implements IPricesApi {
         notFoundExpireTimeSeconds: this.defaultNotFoundExpirationTimeSeconds,
         expireTimeSeconds: this.pricesTtlSeconds,
       });
+      return result?.[nativeCoinId]?.[lowerCaseFiatCode];
     } catch (error) {
-      throw new DataSourceError(
-        `Error getting ${
-          args.nativeCoinId
-        } price from provider${this._mapProviderError(error)}`,
+      this.loggingService.error(
+        `CoinGecko error while getting native coin price: ${asError(error)} `,
```

Comment:
This can also be an error retrieving the `nativeCoinId`.

Reply:
Good point, I've removed the CoinGecko reference, as the error could have a different source as you commented.

[a62b130](https://github.com/safe-global/safe-client-gateway/pull/1175/commits/a62b1305da4626e6b4362831001b84289acfed44)